### PR TITLE
Feature : Move Date Histogram to dimensions

### PR DESCRIPTION
**Pull request by harshada8989**
Signed-off-by: harshada.lingayat <harshada_lingayat@persistent.com>

### Description
Moving the date histogram section inside dimensions panel where it will only render when any timestamp field is selected in dimensions and once it is selected timestamp option will be removed from the dimensions dropdown

### Issues Resolved
#1079
#1141

image

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx [hunk 1]
```diff
@@ -118,12 +129,22 @@ export const DataConfigPanelItem = ({
...
-    const updatedList = { ...list, [name]: arr };
-    setConfigList(updatedList);
+    let list = { ...configList };
+    const isTimeStamp = name === SPAN;
+    if (isTimeStamp) {
+      delete list[SPAN];
+    } else {
+      const arr = [...list[name]];
+      arr.splice(index, 1);
+      list = { ...list, [name]: arr };
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx:145]: Take this as a future task to create an utility function to insert item at a specific position. I see this type of code so many times, those can be updated with this utility function. Make sure to de-structure the elements inside the function and not while calling it. You call the function with an array and the function is going to return you a new array using the supplied list.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx [hunk 2]
```diff
@@ -219,22 +260,32 @@ export const DataConfigPanelItem = ({
     (visualizations.vis.name === VIS_CHART_TYPES.Line ||
       visualizations.vis.name === VIS_CHART_TYPES.Scatter);
 
+  const getTimeStampFilteredFields = (options: IField[]) =>
+    filter(options, (i) => i.type !== TIMESTAMP);
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: See if it is possible to add type to variables.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx [hunk 3]
```diff
@@ -219,22 +260,32 @@ export const DataConfigPanelItem = ({
...
+      visualizations.vis.name === VIS_CHART_TYPES.Scatter
+    )
+      return filter(unselectedFields, (i) => i.type === TIMESTAMP);
+    if (!isTimeStampSelected && configList.span?.time_field.length > 0)
+      return getTimeStampFilteredFields(unselectedFields);
+    if (
+      (selectedConfigItem.name === GROUPBY && selectedConfigItem.index === 0) ||
+      isTimeStampSelected
+    )
+      return unselectedFields;
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: Make sure to check all conditions are working properly. You might have already done so, but make sure we do not skip any conditions here during testing.

**Review by spattnaik — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx (hunk 1 above)
Comment (harshada8989) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx:145]: will take it in next PR and change all across the code where same logic has been implemented

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx (hunk 2 above)
Comment (harshada8989) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: Done

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx (hunk 3 above)
Comment (harshada8989) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: Yes checked it

**Review by ruchika-narang — APPROVED** [review]

**Pull request closed without merging**